app/views/shared/_client_filters.html.erb

```diff
@@ -10,9 +10,9 @@
         <select name="status" class="select__element" id="status-filter">
           <option value></option>
           <% TaxReturnStatus.available_statuses_for(current_user).each do |stage, statuses| %>
-            <option value="<%= stage %>" <%= params[:status] == stage && "selected" %>><%= TaxReturnStatusHelper.stage_translation(stage) %></option>
+            <option value="<%= stage %>" <%= @filters[:stage] == stage && "selected" %>><%= TaxReturnStatusHelper.stage_translation(stage) %></option>
             <% statuses.each do |status| %>
-              <option value="<%= status %>" <%= params[:status] == status.to_s && "selected" %>>&emsp;<%= status_translation(status) %></option>
+              <option value="<%= status %>" <%= [status, status.to_s].include?(@filters[:status]) && "selected" %>>&emsp;<%= status_translation(status) %></option>
```

Comment:
I see we're checking for either a value (int? symbol?) or its stringified version here.

I'd rather see `@filters[:status]` always have either the value or the stringified value because I like knowing exactly what data to look for.

Not a blocker.